examples/llm/txt2kg_rag.py

```diff
@@ -536,7 +578,15 @@ def eval(question: str, pred: str, correct_answer: str):
         print("Error: wandb package not found but --wandb flag was used.")
         print("Please install wandb and rerun the script.")
         sys.exit(1)
+
     print("Starting TechQA training with args: ", args)
-    data_lists = make_dataset(args)
+    if os.path.exists("tech_qa.pt") and not args.regenerate_dataset:
+        print("Re-using Saved TechQA KG-RAG Dataset...")
+        data_lists = torch.load("tech_qa.pt", weights_only=False)
```

Comment:
can we add a check that the vector retriever k doesnt match the current asked k fi we are going to update. if k is the same we dont need to. otherwise lgtm ill merge